### DISCUSSION TITLE
What does the lock timer do?

### DISCUSSION THREAD

**Discussion by trymeouteh** [hardalgorithm/ParanoidsPal-PrivacyLock#3] [Q&A] [answered]
What does the lock timer do that is different from the lock timer in the android settings for when you turn off your screen?

**Comment by hardalgorithm** [comment]
Hi [trymeouteh](https://github.com/trymeouteh) 👾
If you use fingerprint unlock, the app will disable the fingerprint after the timer passes and you can unlock the device with pin code or password only
This is used more for “convenient privacy” in cases when you know that for five minutes you will definitely turn on the phone, but when not using the device for five minutes - it would be better to have it locked

**Reply by trymeouteh** [reply]
This feature will allow you to unlock your phone within the time period with your fingerprint and not force the user to have to  use a PIN/password?

**Reply by hardalgorithm** [reply, marked as answer]
Yes you are right 
If the timer time has not passed the user uses the fingerprint to unlock the phone

**Reply by Taxit0m** [reply]
Isn't fingerprint more secure?  I'm my opinion if someone who isn't me ends up with my unlock pin or password by say eaves dropping in the screen when I unlock the phone i would want to force the fingerprint after an allotted amount of time after the screen locks.   
  The majority of unauthorized unlocks are when after the pin has been snooped and the phone is sitting unattended without the owner full attention, like when laced on the charger, intended snoop removed it from its charging place to snoop the phone because logical thought tells us that's when the easiest window occurs.  Phone s left out of direct attention of the owner for long periods of time when on charge, that's when I would want the fingerprint to be forced and the pin or password disabled.

**Reply by hardalgorithm** [reply]
It might be worth dealing with a situation where a super-secure fingerprint is used against your will or when you are unconscious.

**Reply by Taxit0m** [reply]
The fact that you can opt out of using and storing biometrics and disabling finger print all together if the user is worried about the use of their fingerprint when unconscious would be 100% safer than installing an application that requires the use of a permission granting the developer remote administration privileges and trusting the developer with the sensitive said data they fear may be stolen while unconscious.  In one scenario the user sacrifices a functionality in order to protect their sensitive data, and one scenario the user gives the developer his sensitive data willingly under the masquerade they are installing an application to reduce the risk of sensitive data leak.

**Reply by hardalgorithm** [reply]
Administrative privileges are required to automatically disable biometrics (only _force-lock_ )
**device_admin_receiver.xml**
```
<?xml version="1.0" encoding="utf-8"?>
<device-admin>
    <uses-policies>
        <force-lock />
    </uses-policies>
</device-admin>
```
Unfortunately this is how the api works, you can check the source code and make sure that no data is collected or transmitted.

**Reply by ataraxyarcane** [reply]
@Taxit0m The use case for this app is more nuanced than a hard line between either completely relying on biometrics to unlock your phone and disabling it altogether. The functionality of this app brings the two sides to a middle ground where you can enjoy the convenience and decent security of unlocking your phone with your biometrics while being able to easily fall back to good 'ol pin/password (ideally with a limited number of attempts!) when it's riskier.
